### DISCUSSION TITLE
Add OnClientConnected and OnClientDisconnected methods to NetworkBehaviour

### DISCUSSION THREAD

**Discussion by AlphaLul** [FirstGearGames/FishNet#251] [Feature Request]
Currently, to do something on the server whenever a client connects or disconnects, you have to subscribe a method to the ServerManager.OnRemoteConnectionState event and then check if it is a connection or a disconnection. 

It would be useful to have a simple OnClientConnected and OnClientDisconnected method in the NetworkBehaviour class that runs on the server whenever a client connects or disconnects respectively. It could have a NetworkConnection as a parameter to give info about the newly connected or disconnected client. It should essentially function the same as the current ServerManager.OnRemoteConnectionState event, but be separated into connected/disconnected and built into the NetworkBehaviour class for simplicity.

This would be super useful in use cases such as running a TargetRPC whenever a client joins to provide the usernames of players that have already joined, or removing a client from a team when the client leaves, stuff like that.

**Comment by FirstGearGames** [comment]
Already exists as OnSpawnServer

**Reply by AlphaLul** [reply]
That's not really the same since it runs after the client's object has spawned rather than when the client connects. This makes it so if you need to update nametags or team colors when someone joins, there is a small gap before it happens when all names or colors are default. I'm not sure if that's how OnRemoteConnectionState works, but it would be useful to have something run at the same time as OnStartServer rather than OnStartClient.

**Reply by FirstGearGames** [reply]
It's called when the object is spawned or despawned for that client. When the client connects is irrelevant if they do not have view of the object. OnSpawnServer is probably what you want. If you send a TargetRpc there it will arrive during the same packet as the spawn and be applied immediately.
